domain/bakerystorage/service/config.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+
+	"github.com/go-macaroon-bakery/macaroon-bakery/v3/bakery"
+	"github.com/juju/errors"
+)
+
+type ConfigState interface {
+	InitialiseBakeryConfig(ctx context.Context, localUsersKey, localUsersThirdPartyKey, externalUsersThirdPartyKey, offersThirdPartyKey *bakery.KeyPair) error
```

Comment:
**suggestion:** Having 4 parameters with the same types in an interface isn't quite maintainable (the risk to mess up the implementation is quite high).

Maybe introduce a struct `BakeryConfigKeys` to disambiguate ?

**todo(spellcheck):** In english, doesn't `initialise` is spelled `initialize` with à Z ?

Reply:
> **suggestion:** Having 4 parameters with the same types in an interface isn't quite maintainable (the risk to mess up the implementation is quite high).
> 
> Maybe introduce a struct `BakeryConfigKeys` to disambiguate ?

I was debating creating a config struct here. Will add
 
> **todo(spellcheck):** In english, doesn't `initialise` is spelled `initialize` with à Z ?

Not in real English :gb: :gb: :gb: 

(Canonical has a fun policy in it's style guide which suggests we use British English spelling where possible fyi)

Reply:
Ok. But initialise is apparently loozing the battle.

https://sapling.ai/usage/initialize-vs-initialise

Anyway, let'z defend britizh englizh !